packages/@aws-cdk/aws-logs/test/loggroup.test.ts

```diff
@@ -454,6 +454,26 @@ describe('log group', () => {
       LogGroupName: 'my-log-group',
     });
   });
+
+  test('set data protection policy', () => {
```

Comment:
This needs a lot more test coverage for the various valid inputs.

Reply:
Added test coverage for the other parameters and to fully test the structure of the policy.